### PR TITLE
Update fee display inside publish form

### LINKED ISSUE #1109: Update fee display in publish form
The fees in the pricing tab are outdated. Will update with more details
image

Basically the static fees are the ones in the appConfig, the ones that start with publishMarket...

opc fees will be fetched from the subgraph https://github.com/oceanprotocol/ocean-subgraph/pull/337

### PR DESCRIPTION
**Pull request by KatunaNorbert**
Fixes #1109.

### CODE REVIEW

File: src/@utils/subgraph.ts [hunk 1]
```diff
@@ -238,6 +250,26 @@ export async function fetchDataForMultipleChains(
   return datas
 }
 
+export async function getOpcFeees() {
+  let opcFees
+  const variables = {
+    id: 1
+  }
+  const context = getQueryContext(4)
```
Comment (kremalicious) [inline src/@utils/subgraph.ts]: what's the hardcoded `chainId` doing there? In publish form, need to switch that dynamically based on user wallet

Comment (KatunaNorbert) [reply src/@utils/subgraph.ts]: Right, I realised after making it ready for review and changed it

File: src/components/Publish/Pricing/Fees.tsx [hunk 2]
```diff
@@ -37,6 +46,14 @@ export default function Fees({
   pricingType: 'dynamic' | 'fixed'
 }): ReactElement {
   const [field, meta] = useField('pricing.swapFee')
+  const [opcFees, setOpcFees] = useState<OpcFeesData>(undefined)
+  const { chainId } = useWeb3()
+
+  useEffect(() => {
+    getOpcFeees(chainId).then((response: OpcFeesData) => {
```
Comment (kremalicious) [inline src/components/Publish/Pricing/Fees.tsx:53]: without a wallet, `chainId` is not present leading to errors, so falling back to `1` should be more safe in that case: `chainId || 1`

File: src/components/Publish/Pricing/Fees.tsx [hunk 3]
```diff
@@ -37,6 +46,14 @@ export default function Fees({
...
+      setOpcFees(response)
+    })
+  }, [])
```
Comment (kremalicious) [inline src/components/Publish/Pricing/Fees.tsx:56]: you want this to be refetched when `chainId` changes

**Comment by kremalicious** [comment]
we really should have list of fees we actually want to show here. I have no clue which fee names in subgraph refer to the fee names in our app config. Like, there's `consumeFee` in subgraph but in app config we have 3 different consume fees, or 2 different fees collected on order. Which one is it?

File: src/components/Publish/Pricing/Fees.tsx [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+import React, { ReactElement, useEffect, useState } from 'react'
 import Tooltip from '@shared/atoms/Tooltip'
 import styles from './Fees.module.css'
 import { useField } from 'formik'
 import Input from '@shared/FormInput'
 import Error from './Error'
+import {
+  publisherMarketPoolSwapFee,
+  publisherMarketFixedSwapFee
+} from '../../../../app.config'
```
Comment (kremalicious) [inline src/components/Publish/Pricing/Fees.tsx:10]: within components we can use `const { appConfig } = useSiteMetadata()` but I guess in end doesn't really matter

File: src/@utils/subgraph.ts (hunk 1 above)
Comment (kremalicious) [reply src/@utils/subgraph.ts]: haha yes, you pushed the change when I hit submit on this comment, great timing :-)

File: src/components/Publish/Pricing/Fees.tsx [hunk 5]
```diff
@@ -64,12 +81,18 @@ export default function Fees({
           title="Community Fee"
           name="communityFee"
           tooltip={tooltips.communityFee}
+          value={opcFees?.consumeFee || '0'}
```
[the remark this answers is not in the document]
Comment (KatunaNorbert) [reply src/components/Publish/Pricing/Fees.tsx:84]: As I understood, the community fee is a fee that goes to ocean protocol community, not to the market so none of the ones listed above. It should be the `opcFee` I think, but I get your point, it is confusing.

File: src/@utils/subgraph.ts [hunk 6]
```diff
@@ -238,6 +250,26 @@ export async function fetchDataForMultipleChains(
   return datas
 }
 
+export async function getOpcFeees(chainId: number) {
```
Comment (kremalicious) [inline src/@utils/subgraph.ts:253]: I understand fees are tricky and I also want to scream `feeeees!` but let's use only 2 e's here so it reads `getOpcFees`

**Comment by mihaisc** [comment]
Updated https://github.com/oceanprotocol/market/issues/1082 with opc fees

File: src/@utils/subgraph.ts (hunk 6 above)
Comment (KatunaNorbert) [reply src/@utils/subgraph.ts:253]: :)) god one! I'll change it

**Review by mihaisc — APPROVED** [review]

**Pull request merged**